search/search_repository_whitebox_test.go

```diff
@@ -97,47 +91,39 @@ func (s *searchRepositoryWhiteboxTest) setupTestDataSet() ([]SearchTestDescripto
 		{
 			// will test behaviour when null fields are present. In this case, "system.description" is nil
 			fields: map[string]interface{}{
-				workitem.SystemTitle:     "test nofield sbose title '12345678asdfgh'",
-				workitem.SystemCreator:   "sbose78",
-				workitem.SystemAssignees: []string{"pranav"},
-				workitem.SystemState:     "closed",
+				workitem.SystemTitle: "test nofield sbose title '12345678asdfgh'",
+				workitem.SystemState: "closed",
 			},
 			searchString:   `sbose nofield `,
 			minimumResults: 1,
 		},
 		{
 			// will test behaviour when null fields are present. In this case, "system.description" is nil
 			fields: map[string]interface{}{
-				workitem.SystemTitle:     "test should return 0 results'",
-				workitem.SystemCreator:   "sbose78",
-				workitem.SystemAssignees: []string{"pranav"},
-				workitem.SystemState:     "closed",
+				workitem.SystemTitle: "test should return 0 results'",
+				workitem.SystemState: "closed",
 			},
 			searchString:   `negative case `,
 			minimumResults: 0,
 		}, {
 			// search stirng with braces should be acceptable case
 			fields: map[string]interface{}{
-				workitem.SystemTitle:     "Bug reported by administrator for input = (value)",
-				workitem.SystemCreator:   "pgore",
-				workitem.SystemAssignees: []string{"pranav"},
-				workitem.SystemState:     "new",
+				workitem.SystemTitle: "Bug reported by administrator for input = (value)",
+				workitem.SystemState: "new",
 			},
 			searchString:   `(value) `,
 			minimumResults: 1,
 		}, {
 			// search stirng with surrounding braces should be acceptable case
 			fields: map[string]interface{}{
-				workitem.SystemTitle:     "trial for braces (pranav) {shoubhik} [aslak]",
-				workitem.SystemCreator:   "pgore",
-				workitem.SystemAssignees: []string{"pranav"},
-				workitem.SystemState:     "new",
+				workitem.SystemTitle: "trial for braces (pranav) {shoubhik} [aslak]",
+				workitem.SystemState: "new",
 			},
 			searchString:   `(pranav) {shoubhik} [aslak] `,
 			minimumResults: 1,
 		},
 	}
-	fxt := tf.NewTestFixture(s.T(), s.DB, tf.Identities(2), tf.WorkItems(len(testDataSet), func(fxt *tf.TestFixture, idx int) error {
+	fxt := tf.NewTestFixture(s.T(), s.DB, tf.Identities(2, tf.SetIdentityUsernamesFromString([]string{"creator", "assignee"})), tf.WorkItems(len(testDataSet), func(fxt *tf.TestFixture, idx int) error {
```

Comment:
Nice try :). Notice that all but the last two WIs, all belong to sbose78 and in your change they all belong to one entity. Please see my longish comment on my first idea how to fix it and my second attempt on why even that is bad: https://github.com/fabric8-services/fabric8-wit/pull/1677#pullrequestreview-67969319

Reply:
why does it cause any issue ? we don't search by creator or assignee anyways

Reply:
Okay, then. I didn't look at the test itself but only at the test data. And you didn't refactor it 1:1, which was the only cause of my complaint ;)